RecoTracker/DisplacedRegionalTracking/plugins/DisplacedRegionSeedingVertexProducer.cc

```diff
@@ -93,16 +94,20 @@ void DisplacedRegionSeedingVertexProducer::produce(edm::StreamID streamID,
   const auto &trackClusters = event.get(trackClustersToken_);
 
   // Initialize distances.
-  list<DisplacedVertexCluster> pseudoROIs;
-  list<Distance> distances;
+  vector<DisplacedVertexCluster> pseudoROIs;
+  pseudoROIs.reserve(std::min(maxPseudoROIs_, trackClusters.size()));
+  vector<Distance> distances;
   const double minTrackClusterRadius = minRadius_ - rParam_;
   for (unsigned i = 0; i < trackClusters.size(); i++) {
     const reco::VertexCompositeCandidate &trackCluster = trackClusters[i];
     const math::XYZVector x(trackCluster.vertex());
     if (minRadius_ < 0.0 || minTrackClusterRadius < 0.0 || (x - bs).rho() > minTrackClusterRadius)
       pseudoROIs.emplace_back(&trackClusters.at(i), rParam_);
+    if (pseudoROIs.size() == maxPseudoROIs_)
+      break;
```

Comment:
a `LogWarning` may be useful here to help debugging the outliers.

Reply:
Added